src/utils.js

```diff
@@ -59,12 +59,12 @@ export function merge(...args: Array<?Object>): Object {
   let subject = null
 
   for (var i = 0, len = args.length; i < len; i++) {
-    copy = copy || args[i]
-
-    if (copy == null) {
+    if (isObject(args[i]) === false) {
       continue
     }
```

Comment:
Caught `merge(true, true, {})` would return true. This is because `||` was using the `true` value, and no keys in the object meant that the object casting process would never happen.